### ISSUE TITLE
Add maxRequestSize env var to merlin-server to set Javalin HTTP config

### ISSUE THREAD

**Issue by camargo** [NASA-AMMOS/aerie#515]
Users are running into a case where they need to upload data that is larger than the accepted current default size for Javalin. This ticket is to expose an environment variable to [merlin-server](https://github.com/NASA-AMMOS/aerie/blob/develop/merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/AerieAppDriver.java#L97) so users can set the [Javalin HttpConfig maxRequestSize](https://javalin.io/documentation#httpconfig) based on their use case. We should make the default larger than 1 Mb. We should test a large payload (> 10 Mb) works after this update.

For example when using the [addExternalDataset action](https://github.com/NASA-AMMOS/aerie/blob/develop/deployment/hasura/metadata/actions.yaml#L2), Hasura makes a POST request to the `merlin-server` `/addExternalDataset` endpoint. The default value for the `maxRequestSize` config is [1 Mb](https://github.com/javalin/javalin/blob/master/javalin/src/main/java/io/javalin/config/HttpConfig.kt#L9) and external datasets will often be much larger than 1 Mb. Payloads larger than 1 Mb returns a [413 Payload Too Large](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413).

Additionally we should investigate if there are env vars on the Hasura side to support larger payloads. What even is the upper limit of payload size for Hasura actions? I can't find this documented anywhere.

More questions to think about when working on this ticket are:

- Can we support streaming very large payloads into a Hasura action and  into `merlin-server`?
- If we cannot support streaming, can we support submitting data in batches? This would involve either updating the `addExternalDataset` action to accept batches, or adding an `updateExternalDataset` action to include batch semantics.

See the [internal Slack thread here](https://jpl.slack.com/archives/C0163E42UBF/p1670259156122779).

**Comment by Twisol** [comment]
As a thought, would it be difficult to set up a sidecar API for managing file uploads? I think this alternative was raised quite some time ago (over a year ago, certainly), but it didn't seem to be necessary at the time. I remember reading articles on this back then, like [this one](https://dev.to/hasurahq/building-file-upload-downloads-for-your-hasura-app-3225).

The proposed scheme was something like:
* Create a small service that manages uploads (creation, deletion, download) on the filesystem.
* Add an `uploads` table to Postgres with `id` and `path` columns (at minimum).
* Change Hasura queries that reference paths directly to reference `uploads` IDs instead.

Mostly, the rationale here is, if finagling upload files through Hasura is painful, maybe it's the wrong abstraction for the use-case.

**Comment by camargo** [comment]
@Twisol This is not for file uploads actually. The [gateway](https://github.com/NASA-AMMOS/aerie-gateway) still handles file uploads using the scheme you describe. The `addExternalDataset` action does not accept files, but instead just a (potentially) very large JSON blob which then writes to the [plan_dataset.sql](https://github.com/NASA-AMMOS/aerie/blob/develop/merlin-server/sql/merlin/tables/plan_dataset.sql) table (among others).

**Comment by Twisol** [comment]
Ahh, I see. 🤔 Would it be improper to use the file upload infrastructure here? Users could upload external datasets as files (CSV?), then batch-load them through a custom action. In fact, Postgres has support for reading tables from CSV via [COPY FROM](https://www.postgresql.org/docs/current/sql-copy.html), so maybe there are also some performance and simplicity gains in this direction.

**Comment by camargo** [comment]
@Twisol Yeah it's definitely a possibility! In fact I think users would love being able to just add a raw CSV as an external dataset since that's the main format they are using. It's a little out of scope for this ticket tho since it would require an API change first (refactoring `addExternalDataset`). It would also slightly complicate the API since two requests would be required to create an external dataset instead of one. I'll bring this up when we bring this ticket into a sprint tho and we can decide to make an additional ticket then.

**Comment by Twisol** [comment]
For sure :+1: This issue definitely seems to be focusing on the proximal pain point, I get that. Just trying to think about what an ideal fix might be, aside from any immediate patch 😃

**Comment by camargo** [comment]
Yeah I appreciate the additional perspective!

**Comment by parkerabercrombie** [comment]
Being able to upload CSV directly would be nice. Turning CSV into GraphQL greatly increases the payload size.

Long term I think we will need a more robust solution than just a control on the timeout (especially if we continue using GraphQL). The files we tested with were for a one day plan, with samples every 100 seconds. Even increasing the limit to 10M would only get us to 10 day plans at this sample rate, and we need to be able to support much longer timelines.
